ykcompile/src/lib.rs

```diff
@@ -208,23 +209,87 @@ impl<TT> TraceCompiler<TT> {
                     _ => todo!("{:?}", ty),
                 },
                 Projection::Deref => {
-                    // FIXME Dereferencing a reference to a copyable struct/tuple copies it to the
-                    // stack. So this may also return a memory location.
+                    // FIXME We currently assume Deref is only called on Refs.
+
+                    // Are we dereferencing a reference, if so, what's its type.
+                    let tyid = match ty {
+                        Ty::Ref(rty) => rty.clone(),
+                        _ => todo!(),
+                    };
+
+                    if Self::can_live_in_register(&tyid) {
```

Comment:
I don't think it's true that because a type *could* live in a register, that it has to. For example, if we are out of registers, then we would spill.

Should this check be looking at the `Location` of the instance, rather than whether its type could fit in a register?

Reply:
This check is only to differentiate things like `u8`s from arrays, structs, etc.